frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import { Meta, Story } from "@storybook/react";
+
+import { FromArray } from "@common/types/utilityTypes";
+
+import { action } from "@storybook/addon-actions";
+import { fakeUsers } from "@common/fakeData";
+import BasicTable from "./BasicTable";
+import TableHeader from "./TableHeader";
+import TableFooter from "./TableFooter";
+import {
+  ColumnsOf,
+  handleColumnHiddenChange,
+  IdType,
+  SortingRule,
+} from "./basicTableHelpers";
+import { User } from "../../api/generated";
+
+type ApiManageTableProps = {
+  header: Omit<
+    React.ComponentProps<typeof TableHeader>,
+    | "onSearchChange"
+    | "onColumnHiddenChange"
+    | "hiddenColumnIds"
+    | "filterButtonComponent"
+    | "columns"
+  >;
+};
+
+const mockUsers = fakeUsers(100);
+
+type Users = Array<User>;
+
+type SearchState = {
+  term?: string;
+  type?: string;
+};
+
+type Data = NonNullable<FromArray<Users>>;
+
+const columns: ColumnsOf<Data> = [
+  {
+    label: "First Name",
+    id: "name",
+    accessor: (row: User) => row.firstName,
+    sortColumnName: "title",
+  },
+  {
+    label: "Last Name",
+    id: "name",
+    accessor: (row: User) => row.lastName,
+    sortColumnName: "title",
+  },
+  {
+    label: "Email",
+    id: "email",
+    accessor: (row: User) => row.email,
+    sortColumnName: "email",
+  },
+  {
+    label: "Edit",
+    id: "edit",
+    accessor: () => <a href="/#">Edit</a>,
+  },
+];
+
+export default {
+  component: BasicTable,
+  title: "API Managed Table",
+  subcomponents: {
+    TableHeader,
+    TableFooter,
+  },
+  args: {
+    header: {
+      addBtn: {
+        label: "Add row",
+        path: "/#",
+      },
+    },
+  },
+} as Meta<ApiManageTableProps>;
+
+const allColumnIds = columns.map((c) => c.id);
+
+const Template: Story<ApiManageTableProps> = (args) => {
+  const { header } = args;
+  const [sortingRule, setSortingRule] = React.useState<SortingRule<Data>>();
+  const [hiddenColumnIds, setHiddenColumnIds] = React.useState<IdType<Data>[]>(
+    [],
+  );
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [pageSize, setPageSize] = React.useState(10);
+  const [, setSearchState] = React.useState<SearchState>({
+    type: "",
+    term: "",
+  });
+
+  const paginatedUsers = React.useMemo(() => {
+    const page = currentPage - 1;
+    const start = page * pageSize;
+    const end = start + pageSize;
+    return mockUsers.slice(start, end);
+  }, [pageSize, currentPage]);
+
+  return (
+    <>
+      <h2 id="heading">API Managed Table</h2>
+      <p>Note: Search not functional</p>
+      <TableHeader
+        columns={columns}
+        hiddenColumnIds={hiddenColumnIds}
+        filterButtonComponent={null}
+        onSearchChange={(term, type) => {
+          action(JSON.stringify({ term, type }));
```

Comment:
I don't think I'm getting anything out of this.  Are you?

Reply:
Did not realize I had to wrong syntax. I think we may need to review our story actions cause I just copied an existing one.

Fixed in [fix search action logging](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/b50f888452f8ed6b8e7ee73da94f5fbf91fd4d8c)